ports/esp32/machine_touchpad.c

```diff
@@ -43,16 +43,27 @@
 #include "driver/touch_pad.h"
 #elif SOC_TOUCH_SENSOR_VERSION == 2 // All other SoCs with touch, to date
 #include "driver/touch_sensor.h"
+#elif SOC_TOUCH_SENSOR_VERSION == 3 // All other SoCs with touch, to date
+#include "driver/touch_sens.h"
 #else
 #error "Unknown touch hardware version"
 #endif
 
 typedef struct _mtp_obj_t {
     mp_obj_base_t base;
     gpio_num_t gpio_id;
+#if SOC_TOUCH_SENSOR_VERSION == 1 || SOC_TOUCH_SENSOR_VERSION == 2
     touch_pad_t touchpad_id;
+#elif SOC_TOUCH_SENSOR_VERSION == 3
+    int touchpad_id;
+#endif
 } mtp_obj_t;
-
+#if SOC_TOUCH_SENSOR_VERSION == 3
+static touch_sensor_handle_t  s_sens_handle = NULL;
+static touch_channel_handle_t s_chan_handle[15] = {};
+uint32_t benchmark[14] = {};
+int i;
```

Comment:
These two variables look like they can be local variables inside functions, not globals.